### PR TITLE
Add delete resource command

### PR DESCRIPTION
**Pull request by 16yuki0702**
# Changes
Add deleting the pipelineresource feature for https://github.com/tektoncd/cli/issues/193.
Make PR not too big, I started only pipelineresource.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
you can delete pipelineresource from tkn.
```

### CODE REVIEW

File: pkg/cmd/pipelineresource/delete.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	eg := `
+# Delete a PipelineResource of name 'foo' in namespace 'bar'
+tkn resource delete foo -n bar
+
+tkn res del foo -n bar",
+`
+
+	c := &cobra.Command{
+		Use:          "delete",
+		Aliases:      []string{"del"},
```
Comment (sthaha) [inline pkg/cmd/pipelineresource/delete.go]: how about `rm` as well to match `ls` in `list`

File: pkg/cmd/pipelineresource/delete.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+func deleteResource(s *cli.Stream, p cli.Params, preName string) error {
+	cs, err := p.Clients()
+	if err != nil {
+		return fmt.Errorf("Failed to create tekton client")
+	}
+
+	if err := cs.Tekton.TektonV1alpha1().PipelineResources(p.Namespace()).Delete(preName, &metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("Failed to delete pipelineresource %q", preName)
```
Comment (sthaha) [inline pkg/cmd/pipelineresource/delete.go]: we may want to provide a bit more details as to why the deletion failed by printing the `err` as well.

File: pkg/cmd/pipelineresource/delete.go (hunk 1 above)
Comment (16yuki0702) [reply pkg/cmd/pipelineresource/delete.go]: Thank you very much!!
I didn't realize the list aliases are like that, I'll match it.

File: pkg/cmd/pipelineresource/delete.go (hunk 2 above)
Comment (16yuki0702) [reply pkg/cmd/pipelineresource/delete.go]: Totally agree, thanks!!

**Comment by 16yuki0702** [comment]
@sthaha Thank you for your review! 
I fixed it and add annotations like this PR https://github.com/tektoncd/cli/pull/226

**Comment by 16yuki0702** [comment]
@sthaha oh... Thank you so much! I'll add the documents.

**Comment by 16yuki0702** [comment]
@sthaha I added the documents, thanks!!
when I run ```make docs``` and ```make man```, other documents are also modified.
for now I include these into the PR.

**Comment by 16yuki0702** [comment]
@sthaha @vdemeester Thank for reviews!!

**Pull request merged**